Engine/csound_orc_semantics.c

```diff
@@ -1212,10 +1223,10 @@ char* get_arg_string_from_tree(CSOUND* csound, TREE* tree,
 
   while (current != NULL) {
     char* argType = get_arg_type2(csound, current, typeTable);
-    //FIXME - fix if argType is NULL and remove the below hack
+
     if (argType == NULL) {
-      argsLen += 1;
-      argTypes[index++] = cs_strdup(csound, "@");
+      // if we failed to find argType, exit from parser
+      csound->Die(csound, "Could not parse type for argument %s", current->value->lexeme);
```

Comment:
I think this will not work as a solution as it would make live coding very difficult if one makes a type mistake.

Reply:
it just a does a long jump to orcparse as far as I can see and returns from there. What would be the problem?

Reply:
I looked and on error it should jump back to csoundCompileOrcInternal() and return an error from there. Csound should carry on performing. 

"Die()" is the wrong name for that function, it just prints an error and jumps to the last point set.